x-pack/plugins/observability_solution/apm/public/components/alerting/ui_components/alert_details_app_section/alert_details_contextual_insights.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import {
+  createFunctionRequestMessage,
+  createFunctionResponseMessage,
+} from '@kbn/observability-ai-assistant-plugin/public';
+import React, { useCallback } from 'react';
+import dedent from 'dedent';
+
+import { callApmApi } from '../../../../services/rest/create_call_apm_api';
+import {
+  SERVICE_ENVIRONMENT,
+  SERVICE_NAME,
+  TRANSACTION_NAME,
+  TRANSACTION_TYPE,
+} from '../../../../../common/es_fields/apm';
+import { useKibana } from '../../../../context/kibana_context/use_kibana';
+import { AlertDetailsAppSectionProps } from './types';
+
+export function AlertDetailContextualInsights({
+  alert,
+}: {
+  alert: AlertDetailsAppSectionProps['alert'];
+}) {
+  const {
+    services: {
+      observabilityAIAssistant: { ObservabilityAIAssistantContextualInsight },
+    },
+  } = useKibana();
+
+  const getPromptMessages = useCallback(async () => {
+    const { start, end } = getConversationTimeRange(alert);
+
+    const {
+      apmTimeseries,
+      downstreamDependencies,
+      logCategories,
+      serviceList,
+      serviceSummary,
+    } = await callApmApi(
+      'GET /internal/apm/assistant/get_apm_alert_details_context',
+      {
+        signal: null,
+        params: {
+          query: {
+            [SERVICE_NAME]: alert.fields[SERVICE_NAME],
+            [SERVICE_ENVIRONMENT]: alert.fields[SERVICE_ENVIRONMENT],
+            [TRANSACTION_TYPE]: alert.fields[TRANSACTION_TYPE],
+            [TRANSACTION_NAME]: alert.fields[TRANSACTION_NAME],
+            end,
+            start,
+          },
+        },
+      }
+    );
+
+    const serviceName = alert.fields[SERVICE_NAME];
+
+    return [
+      createFunctionRequestMessage({
+        name: 'get_apm_alert_details_context',
+        args: {
+          [SERVICE_NAME]: alert.fields[SERVICE_NAME],
+          [SERVICE_ENVIRONMENT]: alert.fields[SERVICE_ENVIRONMENT],
+          [TRANSACTION_TYPE]: alert.fields[TRANSACTION_TYPE],
+          [TRANSACTION_NAME]: alert.fields[TRANSACTION_NAME],
+        },
+      }).message,
+
+      createFunctionResponseMessage({
+        name: 'get_apm_alert_details_context',
+        content: dedent(
```

Comment:
`content` for a function response needs to be stringified JSON I think, because it is parsed again in some places (hard to differentiate between a plain string and the result of JSON.stringify)